src/transformers/models/idefics2/modeling_idefics2.py

```diff
@@ -1665,6 +1665,9 @@ def prepare_inputs_for_generation(
         num_logits_to_keep=None,
         **kwargs,
     ):
+        # Overwritten -- there are mutually exclusive inputs (if the logic to make `image_hidden_states` take
+        # precedence is moved to the model, we can remove this fn)
+
```

Comment:
I think most VLMs do smth similar when we pass pixel values only in pre-fill stage. I am thinking that for Idefics it can also be a check on `if cache_position[0] == 0` as we don't support multi-turn dialogues. So I am think we can find a way to generalize for VLMs in a subsequent PR :)

About moving the logic to modeling, I think we want to discourage anyone to pass both